_ml-commons-plugin/mcp/mcp-connector.md

```diff
@@ -0,0 +1,251 @@
+}
+```
+
+### Step 4: Execute Agent
+
+Invoke the registered agent with a query. The agent will load tools from both OpenSearch and the filtered MCP server.
+
+```json
+POST /_plugins/_ml/agents/<Agent ID>/_execute
+{
+  "parameters": {
+    "question": "Any weather alerts in Washington",
+    "verbose": true
+  }
+}
+```
+
+**Response:**
+```
+{
+    "inference_results": [
+        {
+            "output": [
+                {
+                    "name": "memory_id",
+                    "result": "MfiZfpYBjoQOEoSH13wj"
+                },
+                {
+                    "name": "parent_interaction_id",
+                    "result": "MviZfpYBjoQOEoSH13xC"
+                },
+                {
+                    "name": "response",
+                    "result": "{\"id\":\"chatcmpl-BRRcdxVjkrKG7HjkVWZVwueJSEjgd\",\"object\":\"chat.completion\",\"created\":1.745880735E9,\"model\":\"gpt-4o-2024-08-06\",\"choices\":[{\"index\":0.0,\"message\":{\"role\":\"assistant\",\"tool_calls\":[{\"id\":\"call_yWg0wk4mfE2v8ARebupfbJ87\",\"type\":\"function\",\"function\":{\"name\":\"get_alerts\",\"arguments\":\"{\\\"state\\\":\\\"WA\\\"}\"}}],\"annotations\":[]},\"finish_reason\":\"tool_calls\"}],\"usage\":{\"prompt_tokens\":201.0,\"completion_tokens\":16.0,\"total_tokens\":217.0,\"prompt_tokens_details\":{\"cached_tokens\":0.0,\"audio_tokens\":0.0},\"completion_tokens_details\":{\"reasoning_tokens\":0.0,\"audio_tokens\":0.0,\"accepted_prediction_tokens\":0.0,\"rejected_prediction_tokens\":0.0}},\"service_tier\":\"default\",\"system_fingerprint\":\"fp_f5bdcc3276\"}"
+                },
+                {
+                    "name": "response",
+                    "result": "[{\"text\":\"\\nEvent: Wind Advisory\\nArea: Kittitas Valley\\nSeverity: Moderate\\nDescription: * WHAT...Northwest winds 25 to 35 mph with gusts up to 45 mph\\nexpected.\\n\\n* WHERE...Kittitas Valley.\\n\\n* WHEN...From 2 PM to 8 PM PDT Tuesday.\\n\\n* IMPACTS...Gusty winds will blow around unsecured objects. Tree\\nlimbs could be blown down and a few power outages may result.\\nInstructions: Winds this strong can make driving difficult, especially for high\\nprofile vehicles. Use extra caution.\\n\"}]"
+                },
+                {
+                    "name": "response",
+                    "result": "There is a Wind Advisory for the Kittitas Valley in Washington. Here are the details:\n\n- **Event:** Wind Advisory\n- **Area:** Kittitas Valley\n- **Severity:** Moderate\n- **Description:** Northwest winds 25 to 35 mph with gusts up to 45 mph expected.\n- **When:** From 2 PM to 8 PM PDT Tuesday.\n- **Impacts:** Gusty winds may blow around unsecured objects, potentially causing tree limbs to fall, and resulting in a few power outages.\n\n**Instructions:** These strong winds can make driving difficult, especially for high-profile vehicles. Use extra caution if you are traveling in the area."
+                }
+            ]
+        }
+    ]
+}
+```
+
+## References
```

Comment:
I see current document generally add links to the content. Do you see any document add `References` to end of the page ?

Reply:
Instead of references, we can name this section "Further reading"